src/nesymres/architectures/bfgs.py

```diff
@@ -1,170 +0,0 @@
-    candidate = candidate.format(constant="constant")
-    
-    c = 0 
-    expr = candidate
-    for i in range(candidate.count("constant")):
-        expr = expr.replace("constant", f"c{i}",1)
-
-
-
-    
-    print('Constructing BFGS loss...')
-
-    if cfg.bfgs.idx_remove:
-        print('Flag idx remove ON, Removing indeces with high values...')
-        bool_con = (X<200).all(axis=2).squeeze() 
-        X = X[:,bool_con,:]
-        # idx_leave = np.where((np.abs(input_batch[:,3].numpy()))<200)[0]
-        # xx = xx[:,idx_leave]
-        # input_batch = input_batch[idx_leave,:]
-
-
-    max_y = np.max(np.abs(torch.abs(y).cpu().numpy()))
-    print('checking input values range...')
-    if max_y > 300:
-        print('Attention, input values are very large. Optimization may fail due to numerical issues')
-
-    diffs = []
-    for i in range(X.shape[1]):
-        curr_expr = expr
-        for idx, j in enumerate(cfg.total_variables):
-            curr_expr = sp.sympify(curr_expr).subs(j,X[:,i,idx]) 
-        diff = curr_expr - y[i]
-        diffs.append(diff)
-    #         breakpoint()
-    # diff = [sp.sympify(example).replace(y,xx[1,i]).replace(x,xx[0,i]).replace(z,xx[2,i])-input_batch[i,-1] for i in range(input_batch.shape[0])]
-    if cfg.bfgs.normalization_o:
-        raise NotImplementedError
-        diff = [x/max_eq for x in diffs]
-        #diff = [sympify(example).replace(y,xx[1,i]).replace(x,xx[0,i]).replace(z,xx[2,i])-input_batch[i,-1]/max_eq for i in range(input_batch.shape[0])]
-    loss = 0
-    cnt = 0
-    if cfg.bfgs.normalization_type == "NMSE": # and (mean != 0):
-        mean_y = np.mean(y.numpy())
-        if abs(mean_y) < 1e-06:
-            print("Normalizing by a small value")
-        loss = (np.mean(np.square(diffs)))/mean_y  ###3 check
-    elif cfg.bfgs.normalization_type == "MSE": 
-        loss = (np.mean(np.square(diffs)))
-    else:
-        raise KeyError
-    
-    print('Loss constructed, starting new BFGS optmization...') 
-
-    # Lists where all restarted will be appended
-    F_loss = []
-    consts_ = []
-    funcs = []
-    symbols = {i: sp.Symbol(f'c{i}') for i in range(candidate.count("constant"))}
-    
-    for i in range(cfg.bfgs.n_restarts):
-        # Compute number of coefficients
-        x0 = np.random.randn(len(symbols))
-        s = list(symbols.values())
-        #bfgs optimization
-        fun_timed = TimedFun(fun=sp.lambdify(s,loss, modules=['numpy']), stop_after=cfg.bfgs.stop_time)
-        if len(x0):
-            minimize(fun_timed.fun,x0, method='BFGS')   #check consts interval and if they are int
-            consts_.append(fun_timed.x)
-        else:
-            consts_.append([])
-        final = expr
-        for i in range(len(s)):
-            final = sp.sympify(final).replace(s[i],fun_timed.x[i])
-        if cfg.bfgs.normalization_o:
-            funcs.append(max_y*final)
-        else:
-            funcs.append(final)
-        
```

Comment:
This wasn't tracked right by git (used `sed` to make changes and replace the file).

Reply:
Consider going back and making this change in a way that allows git to track more easily, _i.e._ just make the changes manually.